sentry-graphql-22/src/main/java/io/sentry/graphql22/SentryInstrumentation.java

```diff
@@ -0,0 +1,163 @@
+      final @NotNull SentrySubscriptionHandler subscriptionHandler,
+      final boolean captureRequestBodyForNonSubscriptions,
+      final @NotNull List<String> ignoredErrorTypes) {
+    this(
+        beforeSpan,
+        subscriptionHandler,
+        new ExceptionReporter(captureRequestBodyForNonSubscriptions),
+        ignoredErrorTypes);
+  }
+
+  @TestOnly
+  public SentryInstrumentation(
+      final @Nullable SentryGraphqlInstrumentation.BeforeSpanCallback beforeSpan,
+      final @NotNull SentrySubscriptionHandler subscriptionHandler,
+      final @NotNull ExceptionReporter exceptionReporter,
+      final @NotNull List<String> ignoredErrorTypes) {
+    this.instrumentation =
+        new SentryGraphqlInstrumentation(
+            beforeSpan, subscriptionHandler, exceptionReporter, ignoredErrorTypes, TRACE_ORIGIN);
+    SentryIntegrationPackageStorage.getInstance().addIntegration("GraphQL-v22");
+    SentryIntegrationPackageStorage.getInstance()
+        .addPackage("maven:io.sentry:sentry-graphql-22", BuildConfig.VERSION_NAME);
+  }
+
+  /**
+   * @param subscriptionHandler can report subscription errors
+   * @param captureRequestBodyForNonSubscriptions false if request bodies should not be captured by
+   *     this integration for query and mutation operations. This can be used to prevent unnecessary
+   *     work by not adding the request body when another integration will add it anyways, as is the
+   *     case with our spring integration for WebMVC.
+   */
+  public SentryInstrumentation(
+      final @NotNull SentrySubscriptionHandler subscriptionHandler,
+      final boolean captureRequestBodyForNonSubscriptions) {
+    this(null, subscriptionHandler, captureRequestBodyForNonSubscriptions);
+  }
+
+  @Override
+  public @NotNull InstrumentationState createState(
+      final @NotNull InstrumentationCreateStateParameters parameters) {
+    return instrumentation.createState();
+  }
+
+  @Override
+  public @Nullable InstrumentationContext<ExecutionResult> beginExecution(
+      final @NotNull InstrumentationExecutionParameters parameters,
+      final @NotNull InstrumentationState state) {
+    final SentryGraphqlInstrumentation.TracingState tracingState =
+        InstrumentationState.ofState(state);
+    instrumentation.beginExecution(parameters, tracingState);
+    return super.beginExecution(parameters, state);
+  }
+
+  @Override
+  public CompletableFuture<ExecutionResult> instrumentExecutionResult(
+      ExecutionResult executionResult,
+      InstrumentationExecutionParameters parameters,
+      final @NotNull InstrumentationState state) {
+    return super.instrumentExecutionResult(executionResult, parameters, state)
+        .whenComplete(
+            (result, exception) -> {
+              instrumentation.instrumentExecutionResultComplete(parameters, result, exception);
+            });
+  }
+
+  @Override
+  public @NotNull InstrumentationContext<ExecutionResult> beginExecuteOperation(
```

Comment:
graphql `Instrumentation.beginExecutionOperation` is annotated with `@Nullable` so we might return `null` here eventhough our annotation says `@NotNull`